src/fast_reader_bseq.c

```
@@ -0,0 +1,73 @@
+    kseq_t *ks = (kseq_t *)ks1_, *ks2 = (kseq_t *)ks2_;
+    int64_t size = 0, m, n;
+    bseq1_t *seqs;
+    m = n = 0; seqs = 0;
+    while (kseq_read(ks) >= 0) {
+        if (ks2 && kseq_read(ks2) < 0) {                 /* 2nd file has fewer */
+            fprintf(stderr, "[W::%s] the 2nd file has fewer sequences.\n", __func__);
+            break;
+        }
+        if (n >= m) {
+            m = m ? m << 1 : 256;
+            seqs = (bseq1_t *)realloc(seqs, m * sizeof(bseq1_t));
```

Cppcheck (2.20.0)

[error] Common realloc mistake

(memleakOnRealloc)